integration-tests/__tests__/deprecated-cli-options.test.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const path = require('path');
+const runJest = require('../runJest');
+
+const dir = path.resolve(__dirname, '../deprecated-cli-options');
+
+jest.mock('jest-config', () => ({
+  deprecationEntries: {
+    cache: () => 'This option is deprecated.',
+  },
+}));
+
+it('Prints deprecation warnings for CLI flags', () => {
+  const {stderr, stdout} = runJest(dir, ['--cache']);
+  expect(stderr).toMatchSnapshot();
```

Comment:
It's sufficient to just do a regex check for `Test Suites: 1 passed, 1 total`.
Snapshot contains changing data (execution time) which breaks on other machines, unless you replace them with something static.

Reply:
Will do.